### ISSUE TITLE
v3.0.0-rc: Ignore settings for dirs & files in XML-Config are not respected when parsing

### ISSUE THREAD

**Issue by oliveratgithub** [phpDocumentor/phpDocumentor#2335]
## Expected Behavior
Using

```
<phpdoc>
…
	<files>
		<directory>www</directory>
		<ignore>mobilezorg/*</ignore>
		<ignore>highlight*</ignore>
	</files>
…
```
should ignore the directory `/www/mobilezorg/` and all its sub-files, so as any files containing `highlight` in the file name when generating the documentation from the source files.

## Actual Behavior
Specifying various ignore options within a custom XML-config file are not applied / ignored, files under ignored folders or with filename matching ignore setting are parsed anyway:

```
ALERT     [app]   Unable to parse file "www/mobilezorg/mobilezorg.inc.php", an error was detected: Syntax error, unexpected '=' on line 458
ALERT     [app]   Unable to parse file "www/js/highlight-js/highlight.pack.js", an error was detected: Syntax error, unexpected '^', expecting T_STRING on line 2
```

## Steps to Reproduce the Problem

  1. Add a some `<ignore>`-flags under the `<files>` configuration in a custom phpDocu XML-Config, matching with actual directories and files in the code directory.
  2. Run the phpDocumentor PHAR using `php ./phpDocumentor.phar -c /path/to-phpdocu/custom_config.xml`
  3. Check the parsing output to see that files and directories which should be ignored have been parsed anyway

## Your environment

* Version used: v3.0.0-rc
* Install method: phpDocumentor.phar
* php version: PHP 7.3.11 (cli) (built: Apr  3 2020 08:00:20) ( NTS )
* Operating system and version (e.g. Ubuntu 16.04, Windows 7): macOS 10.15.5 Beta (19F62f)
* Link to your project: [zorgch/zorg-code/tree/upgrade-php-7.x](https://github.com/zorgch/zorg-code/tree/upgrade-php-7.x)

**Comment by oliveratgithub** [comment]
Could be related to issue #2325

**Comment by oliveratgithub** [comment]
Related issue #2319

**Comment by mvriel** [comment]
Hello @oliveratgithub, thank you for reporting this issue.

This break happened because we changed the way we deal with wildcards in phpDocumentor 3. We now strictly adhere to glob patterns with the globstar extension; in phpDocumentor 2 we interpreted a closing asterisk as 'anything'.

These two interpretations are mutually exclusive, and to fix this I have added a compatibility layer to 'upgrade' the phpDocumentor 2 version to a true glob pattern.

In short, `tests/*` will secretly be converted to `tests/**/*` ;)

When using configVersion 3 (and higher), glob patterns are the default and this conversion will no longer take place.

This fix is pushed to master and the [next build](https://github.com/phpDocumentor/phpDocumentor/actions/runs/96472625) will contain a newer version of the PHAR file to test with.
